From ab371432032952694978a554e331f79c5b803563 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 07 Apr 2021 10:42:58 +0300
Subject: [PATCH] [SW-37437] gaudi: security: add new pci device ids

New pci device ids have been added in gaudi.
This needs to be added in the existing
logic that uses existing pci device ids (0x1000
or 0x1001).

New device ids:
Gaudi secure:      0x1010
Gaudi FP16 secure: 0x1011

Change-Id: If69cc4b972afd906b59af0325b34b4cd040de9da
Signed-off-by: Karthikeyan K <karthikk@habana.ai>
(cherry picked from commit 5e9fcf83fbdb678cfa4b3cae02c873748ce07a45)
---

diff --git a/gaudi/hbm.c b/gaudi/hbm.c
index fa4fbe0..8f7c1cb 100644
--- a/gaudi/hbm.c
+++ b/gaudi/hbm.c
@@ -1050,6 +1050,7 @@
 	for (ch = 0; ch < GAUDI_HBM_CHANNELS; ch++) {
 		switch (dev_id) {
 		case PCI_IDS_GAUDI_HL2000M:
+		case PCI_IDS_GAUDI_HL2000M_SEC:
 			if (ch == 0 || ch == 4) {
 				/* Release PHY RSTN (set bit 1) */
 				RMWREG32(base + ch * 0x1000 + 0x284, 1, 0x2);
@@ -1130,7 +1131,7 @@
 		WREG32(base + ch * 0x1000 + 0x298, ctrl_rddv_sel);
 
 		/* Enable HBM CK, set PL=0, set rdatvld_dly */
-		if (dev_id != PCI_IDS_GAUDI_HL2000M)
+		if ((dev_id != PCI_IDS_GAUDI_HL2000M) && (dev_id != PCI_IDS_GAUDI_HL2000M_SEC))
 			rdatvld_dly = ctrl_rddv_sel + 8;
 		else
 			rdatvld_dly = ctrl_rddv_sel + 10;
